pkgs/development/compilers/gcc/common/configure-flags.nix

```diff
@@ -244,6 +244,9 @@ let
     ++ lib.optionals (langD) [
       "--with-target-system-zlib=yes"
     ]
+    ++ lib.optionals (targetPlatform.isDarwin && targetPlatform.isx86_64) [
```

Comment:
Can you add a small inline comment why it's needed as well? Otherwise LGTM.

Reply:
Copy-pasted from #242202.

Reply:
Thank you!